File: 55-jump-game/jump-game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        furthest=0
        for i in range(len(nums)):
            if i > furthest:
                return False
            furthest=max(furthest,i+nums[i])
        return True
        
            

File: 55-jump-game/jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int numsLength=nums.length;
        int target=numsLength-1;
        for(int i=numsLength-1;i>-1;i--){
           if(i+nums[i]>=target){
                target=i;
           }
        }
        return target == 0;
        
    }
}